main.go

```diff
@@ -8,6 +8,8 @@ import (
 )
 
 const dataContractFileName = "datacontract.yaml"
+const qualitySpecFileName = "datacontract-quality.yaml"
```

Comment:
Shouldn't we stick with the Syntax we already introduced? External files via "$ref:" and also allowing to embed yaml/strings? We cannot expect that all quality definitions are in yaml format. Can I maybe help with that?

Reply:
That one would need a dedicated issue, I guess.

Locally, we may prefer to have distinct specification files (whether YAML or any other format, you are right), each dedicated to the respective parts of the contracts (_e.g._, schema, quality checks, input, output). Those dedicated specification files are natural when the remote data contract cross-references those parts; the `init` command could then pull/download locally all the specification parts in local dedicated specification files, whose file-names would correspond to the remote cross-references.

The question is: what should we do locally when the specification part is inline? A solution is to nevertheless create a local dedicated file replicating that specification part, and leave the data contract untouched.
That way, when using/wrapping external tools (such as Soda or Great Expectations), those latter can use directly the dedicated specification files: the whole keeps simple, straightforward.
We can also replace in the local data contract the specification part by a cross-reference on the local dedicated specification file. It avoids duplication of the specification, but it seems more fragile to me (while being cleaner).

The whole point is to have locally dedicated specification files. As long as we have that, I am fine with the solution :)

Reply:
The thing is, we want our data contract to know about all quality specifications. Ideally, they are even embedded in the file, so everyone looking at the contract, can see what qualities apply to the data. If you allow injecting the quality file from outside, you make the data contract knowing about it obsolete.

Reply:
Would it help to allow multiple quality specifications in the data contract specification?

Reply:
> Would it help to allow multiple quality specifications in the data contract specification?

Yes, it would. When I presented internally (to colleagues) the working flow with data contracts (from the YAML specification in Git to the quality checks with Soda), almost the first thing they asked was "How can we make sure that different users will want different quality standards?"

I then thought that we could specify, in the data contracts, various sets of quality checks, each with some sort of identifiers and potentially some weight/flag mechanism, in order to enable/disable some of those sets and/or to provide a weighted score for all the sets.
To be more specific, we would have something like:
```yaml
quality:
  - basic:
      weight: 0.2
      type: SodaCL
      specification: "$ref: contracts/data-contract-flight-route-quality-basic.yaml"
  - premium1:
      weight: 0.4
      type: SodaCL
      specification: "$ref: contracts/data-contract-flight-route-quality-premium1.yaml"
  - premium2:
      weight: 0.4
      type: SodaCL
      specification: "$ref: contracts/data-contract-flight-route-quality-premium2.yaml"
```

Reply:
I have opened an issue in the specification repo to address this. We can discuss this there.
https://github.com/datacontract/datacontract-specification/issues/12